### PR TITLE
Add executor for all the runners

### PR DESCRIPTION
**Pull request by jdneo**

### CODE REVIEW

File: extension/src/runners/runnerExecutor.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import { TestNGRunner } from './testngRunner/TestNGRunner';
+
+export class RunnerExecutor {
+    private readonly javaHome: string;
+    private readonly context: ExtensionContext;
+    private isRunning: boolean;
+    private preLaunchTask: cp.ChildProcess | undefined;
+    private runnerMap: Map<ITestRunner, ITestItem[]> | undefined;
+
+    constructor(javaHome: string, context: ExtensionContext) {
```
Comment (yaohaizh) [inline extension/src/runners/runnerExecutor.ts]: Use _javaHome, _context to simplify the private local variable declaration

```js
  constructor(private _javaHome: string, private _context: ExtensionContext) {
```

File: extension/src/runners/runnerExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            window.showErrorMessage(`${error}`);
+        } finally {
+            await this.cleanUp();
+        }
+    }
+
+    private async cleanUp(): Promise<void> {
+        try {
+            if (this.preLaunchTask) {
+                await killProcess(this.preLaunchTask);
```
Comment (yaohaizh) [inline extension/src/runners/runnerExecutor.ts]: What happens if the preLaunchTask is already closed?

Comment (jdneo) [reply extension/src/runners/runnerExecutor.ts]: Changed the logic: swallow the error when we kill the process

File: extension/src/runners/runnerExecutor.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                this.preLaunchTask = undefined;
+            }
+
```
Comment (yaohaizh) [inline extension/src/runners/runnerExecutor.ts]: Should swallow here? No matter what happens insider the preLaunchTask, need clean up the resource anyway.

**Pull request merged**